### PR TITLE
[SW-1503] Avoid null on AutoML include & exlude Algos params

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala [hunk 1]
```diff
@@ -107,6 +107,15 @@ class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
       modelSettings)
   }
 
+  private def determineIncludedAlgos(): Array[Algo] = {
+    val bothIncludedExcluded = getIncludeAlgos().filter(algo => getExcludeAlgos().contains(algo))
+    bothIncludedExcluded.foreach { algo =>
+      logWarning(s"Algorithm '$algo' was specified in both include and exclude parameters. " +
+        s"Excluding the algorithm.")
+    }
+    getExcludeAlgos().filter(algo => !bothIncludedExcluded.contains(algo)).map(Algo.valueOf)
```
Comment (mn-mikke) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala]: WDYT about `getExcludeAlgos().diff(bothIncludedExcluded).map(Algo.valueOf)`?

**Comment by jakubhava** [comment]
Yup, by default all algorithms are used

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala [hunk 2]
```diff
@@ -107,6 +107,15 @@ class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
       modelSettings)
   }
 
+  private def determineIncludedAlgos(): Array[Algo] = {
+    val bothIncludedExcluded = getIncludeAlgos().filter(algo => getExcludeAlgos().contains(algo))
```
[the remark this answers is not in the document]
Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala]: nice!

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala]: Good tip! I think it should be  `getIncludeAlgos().diff(bothIncludedExcluded).map(Algo.valueOf)` to get the right results

**Pull request merged**